woocommerce/woocommerce-android
# [Dynamic Dashboard] Extract Jetpack Benefits banner logic to the parent ViewModel

Part of: #11219 


### Description
Before, the logic of the Jetpack benefits banner was coupled with the stats code, with this PR, it was extracted to the DashboardViewModel, as with the dynamic behavior of the dashboard now, having the logic coupled with stats would mean that we won't show the banner when the stats card is hidden.

### Testing instructions
1. Make sure the `DYNAMIC_DASHBOARD` feature flag is enabled.
2. Sign in using a Jetpack CP site.
3. Confirm the jetpack benefits banner is shown.
4. Dismiss it, and confirm the banner gets hidden.
5. Open the system settings, and change the date of the device to 5 days later.
6. Go back to the app.
7. Confirm the banner is shown.
